### PR TITLE
Feat/error pages

### PR DESCRIPTION
**Pull request by silverbackdan**

### CODE REVIEW

File: src/core/cwa.ts [hunk 1]
```diff
@@ -20,35 +20,47 @@ export default class Cwa {
...
-      return data
+      if (process.server) {
+        let referer = ctx.req.headers.referer
+        if (!referer) {
+          referer = ctx.req.socket.encrypted ? 'https' : 'http' +
+            '://' +
+            ctx.req.host +
+            ctx.route.fullPath
+        }
+        Object.assign(requestHeaders, { referer })
```
Comment (soyuka) [inline src/core/cwa.ts:38]: what's this for?

File: src/core/cwa.ts [hunk 2]
```diff
@@ -68,24 +79,27 @@ export default class Cwa {
...
+        resource && this.$storage.setResource({ id: resource['@id'], name: resource['@type'], isNew: false, resource })
         return callback(resource)
       })
   }
 
   public async fetchRoute (path) {
-    const routeData = await this.fetchItem({ path: `/_/routes/${path}`, preload: ['/page/layout/componentCollections/*/componentPositions/*/component', '/page/componentCollections/*/componentPositions/*/component'] })
-    const pageData = await this.fetchItem({ path: routeData.page })
-    const layoutData = await this.fetchItem({ path: pageData.layout })
+    const { page } = await this.fetchItem({ path: `/_/routes/${path}`, preload: ['/page/layout/componentCollections/*/componentPositions/*/component', '/page/componentCollections/*/componentPositions/*/component'] })
```
Comment (soyuka) [inline src/core/cwa.ts:88]: Shouldn't it be `page = await`?

Comment (silverbackdan) [reply src/core/cwa.ts:88]: I actually reverted this anyway, page may be null. There's a hierarchal structure I forgot where there can be a `pageData` resource between `route` and `page` - been playing this morning.

File: src/core/cwa.ts (hunk 1 above)
Comment (silverbackdan) [reply src/core/cwa.ts:38]: I've changed this to a querystring. When fetching a `componentPosition` there is a feature in the API that looks in the `pageData` for a given route to populate from that. It's for dynamic pages like blog articles where the page structure is the same but components/data can vary on each article (for example). But yes, I'll push progress in a mo.

**Pull request merged**